homeassistant/components/icloud/sensor.py

```diff
@@ -0,0 +1,85 @@
+"""Support for iCloud sensors."""
+import logging
+from typing import Dict
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_USERNAME, DEVICE_CLASS_BATTERY
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import IcloudDevice
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up iCloud devices sensors based on a config entry."""
+    username = entry.data[CONF_USERNAME]
+
+    entities = []
+    for device in hass.data[DOMAIN][username].devices.values():
+        if device.battery_level is not None:
+            _LOGGER.debug("Adding battery sensor for %s", device.name)
+            entities.append(IcloudDeviceBatterySensor(device))
+
+    async_add_entities(entities, True)
+
+
+class IcloudDeviceBatterySensor(Entity):
+    """Representation of a iCloud device battery sensor."""
+
+    def __init__(self, device: IcloudDevice):
+        """Initialize the battery sensor."""
+        self._device = device
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._device.unique_id}_battery"
+
+    @property
+    def name(self) -> str:
+        """Sensor name."""
+        return self._device.name + " battery state"
```

Comment:
Please use f-strings instead of string concatenation, as above.

Reply:
Whoops, don't know why I mixed up